lib/teleterm/daemon/mfaprompt.go

```diff
@@ -125,7 +126,7 @@ func (p *mfaPrompt) promptWebauthn(ctx context.Context, chal *proto.MFAAuthentic
 
 func (p *mfaPrompt) promptMFA(ctx context.Context, chal *proto.MFAAuthenticateChallenge, runOpts libmfa.RunOpts) (*proto.MFAAuthenticateResponse, error) {
 	resp, err := p.promptAppMFA(ctx, &api.PromptMFARequest{
-		RootClusterUri: p.clusterURI,
+		RootClusterUri: p.resourceURI.GetRootClusterURI().String(),
```

Comment:
This I'm not 100% sure about. Based on [the modal implementation](https://github.com/gravitational/teleport/blob/d6481fb350d772775cb5212b48516e9509e50dd5/web/packages/teleterm/src/ui/ModalsHost/modals/ReAuthenticate/ReAuthenticate.tsx#L86-L92), this field is used purely for display only, unlike cluster URI in the login modal.

We might ask the user to authenticate an action that's related to a leaf resource. It's also good to know which cluster the given resource belongs to.

OTOH, the authentication step always happens in the root cluster. As in, we shouldn't make the user think that there's a separate set of devices for authenticating in root clusters and a separate set for leaf clusters. Also, we always show the modal after the user performs an action. It shouldn't ever catch someone by surprise. So I suppose they'll have a good idea as to which resource they're trying to access.

I think it'd be best if the modal said something like:

> Verify your identity on [root cluster]
>
> MFA is required to access Kubernetes cluster "minikube" from leaf cluster [leaf cluster]

This would require passing `ClusterUri` instead and adding some conditional logic to the modal. However, I think for now it's fine if we always show the root cluster name in the title of the modal.

For reference, this is how the modal looks like:

<img width="480" alt="verify-identity" src="https://github.com/gravitational/teleport/assets/27113/0aa7aee6-9122-4d24-806c-00c3b5b950cc">

Reply:
>OTOH, the authentication step always happens in the root cluster. As in, we shouldn't make the user think that there's a separate set of devices for authenticating in root clusters and a separate set for leaf clusters.

Yeah, that was my reasoning.

>Also, we always show the modal after the user performs an action. It shouldn't ever catch someone by surprise. So I suppose they'll have a good idea as to which resource they're trying to access.

Actually, it may catch you by surprise. If you had a few tabs open that required per-session MFA , and you reopen the previous session, you will see an MFA pop-up for each of them, one after another. So perhaps I will change `rootClusterUri` to `clusterUri` in the proto message and handle it in the dialog? It doesn't seem to be too much work.

Reply:
Sounds good.